olive/passes/onnx/transformer_optimization.py

```diff
@@ -140,6 +155,25 @@ def _run_for_config(
         for key in get_external_data_config():
             del run_config[key]
 
+        if model.hf_config and model.hf_config.config:
```

Comment:
I would like the relationship between `model_name`, `model_path`, `config` to be better defined first. 
If we want to combine the config from `model_path`/`model_name` and `hf_config`, I think we have to load the `PreTrainedConfig` here correctly instead of just looking at `hf_config.config`. Since the config can also be loaded directly from model name or path. 

What I am thinking is we use the `hf_config.config` as kwargs when calling `from_pretrained` for both model and config load. https://github.com/huggingface/transformers/blob/main/src/transformers/configuration_utils.py#L435 
This meets the description given in the docs. 
However, there might be the case that `model_name` is not present. In which case we can just do one of:
- PretrainedConfig.from_dict`: not reliable since it returns a `PretrainedConfig` which might not have some fields. 
- Get `_name_or_path` from `hf_config.dict` and use it as model_name: probably the best option

Also there is no need to add `hf_config.config` if we already have model_name and model_path is None, in OnnxConversion.

Reply:
This PR https://github.com/microsoft/Olive/pull/520 adds `load_model` and `load_config_model` where we could also handle the above.

Reply:
> Also there is no need to add hf_config.config if we already have model_name and model_path is None, in OnnxConversion.

if we already have model_name and model_path is **not** None?

Reply:
Updated with auto_tune_config as we discussed offline.